corehq/apps/receiverwrapper/views.py

```diff
@@ -61,10 +63,45 @@
 PROFILE_LIMIT = int(PROFILE_LIMIT) if PROFILE_LIMIT is not None else 1
 
 
+def _audit_request_auth_errors(request, user_id):
+    cache_key = f"form_submission_audit:{user_id}"
+    if cache.get(cache_key):
+        return []
+
+    errors = []
+
+    def check_route_invalid():
+        stack = inspect.stack()
+        function_names = [frame.function for frame in stack if not inspect.isbuiltin(frame.function)]
+        if post_api.__name__ not in function_names:
+            errors.append(f"Request not made from {post_api.__name__} handler")
+
+    def check_user_permission_invalid():
+        mobile_endpoint_permission = 'access_mobile_endpoints'
+        if not request.user.has_perm(mobile_endpoint_permission):
+            errors.append(f"Required permission {mobile_endpoint_permission} missing")
+
+    check_route_invalid()
+    check_user_permission_invalid()
```

Comment:
Ok, logically then it will be  `check_route_invalid()` AND `check_user_permission_invalid()`. If it returns `True` then log the user. Is that correct?